models/Asset.php

```diff
@@ -737,7 +737,7 @@ protected function update(array $params = []): void
 
                 // delete old legacy file if exists
                 $dbPath = $this->getDao()->getCurrentFullPath();
-                if ($dbPath !== $path && $storage->fileExists($dbPath)) {
+                if (!empty($dbPath) && $dbPath !== $path && $storage->fileExists($dbPath)) {
```

Comment:
Thank you for your PR!
LGTM, just suggesting

```suggestion
                if ($dbPath && $dbPath !== $path && $storage->fileExists($dbPath)) {
```
see also https://localheinz.com/articles/2023/05/10/avoiding-empty-in-php/ as seen on https://github.com/pimcore/pimcore/pull/14799#discussion_r1279528727